### PR TITLE
Enable libhugetlbfs test to run with available pages

### PR DESCRIPTION
**Pull request by harish-24**
Patch enables test to run even with fewer available pages, which would be helpful with system with lesser memory/larger page size.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
```
 (1/1) memory/libhugetlbfs.py:LibHugetlbfs.test:  PASS (6241.59 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 6241.66 s
```

File: memory/libhugetlbfs.py [hunk 1]
```diff
@@ -89,24 +89,36 @@ def setUp(self):
...
-            genio.write_file(
-                '/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages' %
-                str(int(hp_size) * 1024), str(pages_requested))
+            try:
+                genio.write_file(
+                    '/sys/kernel/mm/hugepages/hugepages-%skB/nr_hugepages' %
+                    str(int(hp_size) * 1024), str(pages_requested))
+            except OSError:
+                if (int(hp_size) * 1024) == 16777216:
+                    self.log.warn('Running 16GB hugepages')
```
Comment (PraveenPenguin) [inline memory/libhugetlbfs.py]: why not cancel here as well

File: memory/libhugetlbfs.py [hunk 2]
```diff
@@ -89,24 +89,36 @@ def setUp(self):
...
                 '/sys/kernel/mm/hugepages/huge'
                 'pages-%skB/nr_hugepages' % str(int(hp_size) * 1024).strip()))
             if pages_available < pages_requested:
-                self.cancel('%d pages available, < %d pages requested'
-                            % (pages_available, pages_requested))
+                self.log.warn('%d pages available, < %d pages '
+                              'requested', pages_available, pages_requested)
+
+            if pages_available:
+                self.hugetlbfs_dir.update({hp_size: tempfile.mkdtemp(dir=self.teststmpdir, prefix='avocado_' + __name__)})
```
Comment (PraveenPenguin) [inline memory/libhugetlbfs.py]: please wrap in 80 Char

File: memory/libhugetlbfs.py (hunk 1 above)
Comment (harish-24) [reply memory/libhugetlbfs.py]: This change is to not cancel and continue with the tests and **16G** is not configurable with Hash page table.

File: memory/libhugetlbfs.py (hunk 2 above)
Comment (harish-24) [reply memory/libhugetlbfs.py]: Done. Thanks.

**Comment by harish-24** [comment]
New changes:
Add option to run functional tests

Add option to run functional tests in libhugetlbfs tests

**Comment by harish-24** [comment]
```
JOB LOG    : /home/harish/avocado-fvt-wrapper/results/job-2020-01-10T15.37-3ad6252/job.log
 (1/1) memory/libhugetlbfs.py:LibHugetlbfs.test;default-hugetlbfs_dir-b47c: PASS (30.81 s)
```

**Pull request merged**